spec/lib/nerve_spec.rb

```diff
@@ -0,0 +1,139 @@
+require 'spec_helper'
+require 'nerve/configuration_manager'
+require 'nerve/service_watcher'
+require 'nerve/reporter'
+require 'nerve'
+
+describe Nerve::Nerve do
+  let(:config_manager) { Nerve::ConfigurationManager.new() }
+  let(:mock_config_manager) { double() }
+  let(:mock_method) { double() }
+  let(:nerve_config) { "#{File.dirname(__FILE__)}/../../example/nerve.conf.json" }
+  let(:nerve_instance_id) { 'testid' }
+
+  describe 'check run' do
+    subject {
+      allow(config_manager).to receive(:parse_options_from_argv!) { {
+        :config => nerve_config,
+        :instance_id => nerve_instance_id,
+        :check_config => true
+      } }
+      config_manager.parse_options!
+      Nerve::Nerve.new(config_manager)
+    }
+
+    it 'starts up and checks config' do
+      expect{subject.run}.not_to raise_error
+    end
+  end
+
+  describe 'full application run' do
+    before(:each) {
+      allow_any_instance_of(Nerve::ServiceWatcher).to receive(:run) {
```

Comment:
in general, `allow_any_instance_of` is a smell. rspec recommends not using it if possible: https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

any reason you can't just instantiate a specific watcher? you could maybe define a `SleepingWatcher` in specs which just implements this function.